devto-draft.md

```diff
@@ -1 +1,393 @@
+
+Example:
+
+```vue
+<!-- Vue Parent Component -->
+<template>
+  <div>
+    <child-component :message="message" />
+  </div>
+</template>
+
+<script>
+import ChildComponent from "./ChildComponent.vue";
+export default {
+  data() {
+    return {
+      message: "hello world"
+    };
+  },
+  components: {
+    ChildComponent
+  }
+};
+</script>
+```
+
+```vue
+<!-- Vue Child Component -->
+<template>
+  <div>
+    {{ message }}
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    message: {
+      type: String,
+      default: undefined
+    }
+  }
+};
+</script>
+```
+
+Vue requires you to declare a child component twice, once on import another on `Components`.
+
+React differs in that there is one less step, you import the file then use it.
+
+## Type Enforcement
+
+One thing I miss in static languages like C# or Java is type enforcement. Javascript is a dynamic scripting language. Meaning it doesn't care if a variable is a string, number, boolean, etc.
+
+When you declare a string, you can always typecast it to a number after. This makes scaling large frontend apps difficult. Because you might not know what data type you're working with.
+
+React natively solves this with `PropTypes`. Example in the `<Child/>` component we made earlier:
+
+```js
+import React from "react";
+import PropTypes from "prop-types";
+
+const Child = props => {
+  return <div>{props.message}</div>;
+};
+
+Child.propTypes = {
+  message: PropTypes.string.isRequired
+};
+
+export default Child;
+```
+
+There are a few issues with propTypes in React
+
+- It's really easy to misspell
+- It's optional
+- You have to declare an import
+
+Because of these reasons, I find myself forgetting to use propTypes. You could opt to use Typescript instead with React, but this just means more additional time configuring your setup.
+
+Vue forces you to declare your props, in a way that makes sense. Inside your `<script>` tag where all the other logic lives
+
+Example for a Vue child component:
+
+```vue
+<template>
+  <div>
+    {{ message }}
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    message: {
+      type: String,
+      default: undefined
+    }
+  }
+};
+</script>
+
+<style lang="scss" scoped></style>
+```
+
+This is a much better design overall and makes it so that Typescript is not necessary in Vue.
+
+## Data Reactivity
+
+This was the concept that tripped me on Vue several times. In React, everything is reactive. Meaning you can add properties to objects, and React would call it's lifecycle when properties changed.
+
+In Vue ... things don't work the same. In React, every time you modify state, you could pass a new entire state object.
+
+Vue differed in that you _mutated_ your state directly through methods. Because we're intrinsically adding side effects, you have to actually _declare_ when new properties are being added to the object.
```

Comment:
```suggestion
In Vue, you _mutate_ your state directly through methods. Because you're intrinsically adding side effects, you have to actually _declare_ when new properties are being added to the object.
```

Ensure that pronounces are always on check and be consistent with their use. Here you were using "we" and "you" in the same sentence.